internal/nodejs-action/generate-layout.sh

```diff
@@ -2,7 +2,35 @@
 
 set -euo pipefail
 
```

Comment:
thoughts on using TS for this code? Maybe as a follow-up?

Reply:
Yeah, I'd like to use TS as much as possible.

It's just easier to use Bash since to use TS I'd need to write yet another GitHub action with all the boilerplate (`action.yml`, `package.json`, `Makefile` etc.)

I could use [`github-script`](https://github.com/actions/github-script) but that suffers from a lot of the same problems as using bash (hard to test, lint etc.)

I added #1849 to track.